### DISCUSSION TITLE
Freedom of motion

### DISCUSSION THREAD

**Discussion by LeXXik** [jrouwe/JoltPhysics.js#167] [Q&A] [answered]
In Ammo there is a linear and angular factors, which specify a factor that is multiplied per axis. For example linear factor of [1, 0, 1] means that the body is allowed to move on X and Z, but not on Y. An angular factor of [0, 1, 0] would mean the body can only rotate on Y axis. What would be the way to achieve similar functionality in Jolt? There is a degree of freedom setting on the body, which is using one of the `EAllowedDOFs`. But I don't see how they can be combined, e.g. only allow translation X and Z and rotation Y.

**Comment by PhoenixIllusion** [comment, marked as answer]
The EAllowedDOF is defined as a 6-bit-flag that has both 3-bits of rotation and 3-bits of translation.

Is that what you are referring to, or is this not accomplishing what you are attempting?

const translationXZ_rotationY = EAllowedDOFs_TranslationX | EAllowedDOFs_TranslationZ | EAllowedDOFs_RotationY

https://github.com/jrouwe/JoltPhysics/blob/7edba88d5336047ebae8c6e26a5aa89dd5ae069e/Jolt/Physics/Body/AllowedDOFs.h#L10
```
None			= 0b000000,									///< No degrees of freedom are allowed. Note that this is not valid and will crash. Use a static body instead.
All			= 0b111111,									///< All degrees of freedom are allowed
TranslationX		= 0b000001,									///< Body can move in world space X axis
TranslationY		= 0b000010,									///< Body can move in world space Y axis
TranslationZ		= 0b000100,									///< Body can move in world space Z axis
RotationX		= 0b001000,									///< Body can rotate around world space X axis
RotationY		= 0b010000,									///< Body can rotate around world space Y axis
RotationZ		= 0b100000,									///< Body can rotate around world space Z axis
Plane2D			= TranslationX | TranslationY | RotationZ,	///< Body can only move in X and Y axis and rotate around Z axis
```

**Reply by LeXXik** [reply]
Yep, that is actually what I want. Would that be correct use in JS?
```
mAllowedDOFs = Jolt.EAllowedDOFs_TranslationX | Jolt.EAllowedDOFs_TranslationZ | Jolt.EAllowedDOFs_RotationY;
```

**Reply by jrouwe** [reply]
That should work yes
